libs/sdk-ui-pivot/src/impl/structure/tableDescriptorFactory.ts

```diff
@@ -124,6 +125,13 @@ function groupColumns(
             const groupLevel = columnGroupLevels[level];
             let currentGroup: ScopeCol | undefined = groupLevel.pkToGroup[pathToGroup];
 
+            // TODO: is there a better way?
+            const numberOfTotalHeaders = attributeHeaders.filter((header) =>
```

Comment:
please consider encapsulating at least to something like 
```
function getTotalInfo(attributeHeaders) {
            const numberOfTotalHeaders = attributeHeaders.filter(isResultTotalHeader(header).length;
            return {
               isTotal: numberOfTotalHeaders > 0 && numberOfTotalHeaders === attributeHeaders.length,
               isSubtotal: numberOfTotalHeaders > 0 && numberOfTotalHeaders !== attributeHeaders.length
            };
}
```
and you can spread the result there later.
Also consider reusing (not sure if possible) as it's there twice.

Reply:
Moved to function.